core-bundle/src/Routing/ContentUrlGenerator.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing;
+
+use Contao\CoreBundle\Routing\Content\ContentUrlResolverInterface;
+use Contao\CoreBundle\Routing\Page\PageRegistry;
+use Contao\CoreBundle\Routing\Page\PageRoute;
+use Contao\Model;
+use Contao\PageModel;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\Mapping\MappingException;
+use Symfony\Cmf\Component\Routing\RouteObjectInterface;
+use Symfony\Component\Routing\Exception\ExceptionInterface;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RequestContextAwareInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+class ContentUrlGenerator implements ResetInterface, RequestContextAwareInterface
+{
+    /**
+     * @var array<string, string>
+     */
+    private array $urlCache = [];
+
+    public function __construct(
+        private readonly UrlGeneratorInterface $urlGenerator,
+        private readonly PageRegistry $pageRegistry,
+        private readonly EntityManagerInterface $entityManager,
+        /**
+         * @var iterable<ContentUrlResolverInterface> $urlResolvers
+         */
+        private readonly iterable $urlResolvers,
+    ) {
+    }
+
+    /**
+     * @throws ExceptionInterface
+     */
+    public function generate(object $content, array $parameters = [], array $optionalParameters = []): string
+    {
+        $cacheKey = sha1(serialize($content).'__'.serialize($parameters).'__'.serialize($optionalParameters));
```

Comment:
```suggestion
        $cacheKey = sha1(serialize($content)."\0".serialize($parameters)."\0".serialize($optionalParameters));
```

Just because this is what we usually do in Symfony core.